### PR TITLE
Refactor chi0 models out of HybridMixing

### PR DESCRIPTION
**Pull request by mfherbst**
A bit of refactoring to make it easier to play with different χ0 models inside HybridMixing. Also implements "exact" χ0 for mixing. Happy to discuss alternatives.

### CODE REVIEW

**Review by antoine-levitt — COMMENTED** [review]
Looks about right. I would maybe just have used plain old functions instead of structs, but it doesn't change much probably

File: src/scf/chi0models.jl [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+    function apply!(δρ, δV)
+        # χ0δV[1] is total, χ0δV[2] is spin
+        χ0δV = apply_χ0(ham, ψ_cvg, εF, eigenvalues_cvg, δV...; χ0.kwargs_apply_χ0...)
+        if basis.model.n_spin_components == 1
+            δρ[1] .+= χ0δV[1].real
+        else
+            δρ[1] .+= (χ0δV[1].real .+ χ0δV[2].real) ./ 2
+            δρ[2] .+= (χ0δV[1].real .- χ0δV[2].real) ./ 2
+        end
```
Comment (antoine-levitt) [inline src/scf/chi0models.jl:103]: OK by now we have three cohabitating conventions in the code: (rhoalpha, rhobeta), (rhotot, rhodiff) and rho[:,:,:,ispin]. Can't we cut it down a bit, preferably to just the second?

Comment (mfherbst) [reply src/scf/chi0models.jl:103]: I agree I think we should, but as it turns out I'm not sure the second is the best. It's not the subject of this PR, but I start to get the feeling that internally using `ρ[:, :, :, σ]` and wrapping that in a convenience datastructure (or just providing convenience functions to get `total` and `spin`) might be best.

Comment (antoine-levitt) [reply src/scf/chi0models.jl:103]: That makes sense, if we decide not to go for a special array type

Comment (mfherbst) [reply src/scf/chi0models.jl:103]: Even then. Whether you "wrap" and AbstractArray or a SpecialArray (which is really a list of 2 arrays) should not matter.

**Pull request merged**